packages/ra-core/src/controller/ListController.tsx

```diff
@@ -222,6 +231,12 @@ export class UnconnectedListController extends Component<Props> {
             nextProps.translate === this.props.translate &&
             nextProps.isLoading === this.props.isLoading &&
             nextProps.version === this.props.version &&
+            nextProps.filter === this.props.filter &&
+            nextProps.params.sort !== this.props.params.sort &&
```

Comment:
Same remark as above, don't add the test for params key as it has nothing to do with the bug you're solving.

Reply:
@fzaninotto in my testing when I removed the params comparison on `shouldComponentUpdate` the list failed to immediately update with no query string present.  (`/#/users`)

This caused the sort filter to have to be clicked twice to be applied.

Reply:
That means you have broken something in your PR that used to work before. Applications built with the current version have no problem displaying a resource list page without query string in the URL.